ts/components/session/LeftPaneMessageSection.tsx

```diff
@@ -438,41 +440,71 @@ export class LeftPaneMessageSection extends React.Component<Props, State> {
     }
   }
 
-  private handleJoinChannelButtonClick(groupUrl: string) {
+  private async handleJoinChannelButtonClick(serverUrl: string) {
     const { loading } = this.state;
 
     if (loading) {
-      return false;
+      return;
     }
 
-    // longest TLD is now (20/02/06) 24 characters per https://jasontucker.blog/8945/what-is-the-longest-tld-you-can-get-for-a-domain-name
-    const regexURL = /(http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,24}(:[0-9]{1,5})?(\/.*)?/;
+    // Server URL entered?
+    if (serverUrl.length === 0) {
+      return;
+    }
 
-    if (groupUrl.length <= 0) {
+    // Server URL valid?
+    if (!OpenGroup.validate(serverUrl)) {
       window.pushToast({
         title: window.i18n('noServerURL'),
-        type: 'error',
         id: 'connectToServerFail',
+        type: 'error',
       });
 
-      return false;
+      return;
     }
 
-    if (!regexURL.test(groupUrl)) {
+    // Already connected?
+    if (Boolean(await OpenGroup.getConversation(serverUrl))) {
       window.pushToast({
-        title: window.i18n('noServerURL'),
+        title: window.i18n('publicChatExists'),
+        id: 'publicChatExists',
         type: 'error',
-        id: 'connectToServerFail',
       });
 
-      return false;
+      return;
     }
 
-    MainViewController.joinChannelStateManager(this, groupUrl, () => {
-      this.handleToggleOverlay(undefined);
+    // Connect to server
+    const successPromise = OpenGroup.join(serverUrl, async () => {
+      if (await OpenGroup.serverExists(serverUrl)) {
+        window.pushToast({
+          title: window.i18n('connectingToServer'),
+          id: 'connectToServerSuccess',
+          type: 'success',
+        });
+
+        this.setState({ loading: true });
+      }
     });
 
-    return true;
+    successPromise
```

Comment:
use `try-catch` here, we try not to use `.then` and `.catch` unless absolutely necessary